src/util/expr.h

```diff
@@ -13,6 +13,7 @@ Author: Daniel Kroening, kroening@kroening.com
 #define OPERANDS_IN_GETSUB
 
 #include "type.h"
+#include <memory>
```

Comment:
If there is a way of avoiding this include in that header file this would be preferable (compilation speed). Options are: 1) move both declarations and definitions to a separate class (as only some code will need to use this); 2) just declare the begin/end functions here, and move all definitions to expr.cpp.